src/spikeinterface/postprocessing/template_metrics.py

```diff
@@ -829,9 +829,13 @@ def exp_decay(x, decay, amp0, offset):
     max_channel_location = channel_locations[np.argmax(peak_amplitudes)]
     channel_distances = np.array([np.linalg.norm(cl - max_channel_location) for cl in channel_locations])
     distances_sort_indices = np.argsort(channel_distances)
-    # np.float128 avoids overflow error
-    channel_distances_sorted = channel_distances[distances_sort_indices].astype(np.float128)
-    peak_amplitudes_sorted = peak_amplitudes[distances_sort_indices].astype(np.float128)
+    channel_distances_sorted = channel_distances[distances_sort_indices]
+    peak_amplitudes_sorted = peak_amplitudes[distances_sort_indices]
+    
+    if hasattr(np, 'float128'):
```

Comment:
Can you use `np.longdouble`? 

https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.longdouble

Check out thet section about extended precision:
https://numpy.org/doc/stable/user/basics.types.html#extended-precision

Reply:
From what I understand np.float128 is an alias for np.longdouble.
I prefer float32, float64, float128 over single, double, longdouble.
This is more easy to read no ?

Reply:
It is more easy to read. 

But it seems that np.longdouble adjust to np.float64 or np.float128 depending on the platform. Check it out, I did this on windows:

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/6ed46e04-1cda-455f-a1bc-584c4f60dcda)

And I did this on my code space virtual machine

```python
@h-mayorquin ➜ /workspaces/spikeinterface (main) $ ipython
Python 3.10.8 (main, Nov  2 2023, 18:07:42) [GCC 9.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.17.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import numpy as np

In [2]: np.ones(3, dtype=np.longdouble)
Out[2]: array([1., 1., 1.], dtype=float128)
```

Seems like safer to rely on numpy to obtain the right largest float instead of doing it ourselves.

Reply:
That's the second link that I shared.

Reply:
OK. I did not get it sorry.
Lets do this.